pyaedt/edb_core/EDB_Data.py

```diff
@@ -944,10 +944,16 @@ def _update_hole_parameters(self, hole_type=None, params=None, offsetx=None, off
             offsety = self.hole_offset_y
         if not rotation:
             rotation = self.hole_rotation
-        newPadstackDefinitionData.SetHoleParameters(hole_type, convert_py_list_to_net_list(params),
-                                                      self._edb_value(offsetx),
-                                                      self._edb_value(offsety),
-                                                      self._edb_value(rotation))
+        if "IronPython" in sys.version or ".NETFramework" in sys.version:
```

Comment:
We can consider having a global attribute within pyaedt that checks if we're within IronPython, so we don't have to keep writing this check everwhere.

within `pyaedt/__init__.py`

```py
USING_IRONPYTHON = "IronPython" in sys.version or ".NETFramework" in sys.version
```
Then used with:
```py
pyaedt.USING_IRONPYTHON
# or
from pyaedt import USING_IRONPYTHON
```

Reply:
Good points. I'll create another PR on that suggestions and remove all the checks